addons/project/models/project_task.py

```diff
@@ -823,6 +829,9 @@ def copy_data(self, default=None):
                     'parent_id': False,
                 }
                 vals['child_ids'] = [Command.create(child_id.copy_data(default)[0]) for child_id in task.child_ids]
+            if vals['user_ids']:
+                    cache_value = self._fields['user_ids'].convert_to_cache(vals.get('user_ids', []), self.env['project.task'])
+                    vals['user_ids'] = [Command.set([user_id for user_id in cache_value if user_id in active_user])]
```

Comment:
a = [1,2]
b = [2,3,1]
set(a) & set(b) 

we have to use set intersection instead of list comprehension.